packages/flutter/lib/src/widgets/navigator.dart

```diff
@@ -88,17 +101,31 @@ class RouteSettings {
   }
 }
 
+/// Creates a route for the given route settings.
 typedef Route RouteFactory(RouteSettings settings);
+
+/// A callback in during which you can perform a number of navigator operations (e.g., pop, push) that happen atomically.
 typedef void NavigatorTransactionCallback(NavigatorTransaction transaction);
 
+/// An interface for observing the behavior of a [Navigator].
 class NavigatorObserver {
   /// The navigator that the observer is observing, if any.
   NavigatorState get navigator => _navigator;
   NavigatorState _navigator;
+
+  /// The [Navigator] pushed the given route.
   void didPush(Route route, Route previousRoute) { }
+
+  /// THe [Navigator] popped the given route.
   void didPop(Route route, Route previousRoute) { }
 }
 
+/// Manages a set of child widgets with a stack discipline.
+///
+/// Many apps have a navigator near the top of their widget hiearchy in order to
+/// manage the top of their visual herarchy. For example, a navigator can manage
+/// a visual transition from one page of an app to another or can show a dialog
+/// above the current page.
```

Comment:
Maybe try and describe Navigator in terms of a history stack, rather than in terms of visual hierarchy?